celery/backends/database/models.py

```diff
@@ -26,7 +25,7 @@ class Task(ResultModelBase):
                    primary_key=True, autoincrement=True)
     task_id = sa.Column(sa.String(155), unique=True)
     status = sa.Column(sa.String(50), default=states.PENDING)
-    result = sa.Column(PickleType, nullable=True)
+    result = sa.Column(sa.LargeBinary(length=(2 ** 31) - 1), nullable=True)
```

Comment:
isn't this a breaking change?

Reply:
Not sure...I believe this could break celery apps using a serializer other than pickle when the results backend is sqlalchemy based.
According to the sqlalchemy docs:

> PickleType builds upon the Binary type to apply Python’s pickle.dumps() to incoming objects, and pickle.loads() on the way out, allowing any pickleable Python object to be stored as a serialized binary field.

So, the database column should stay binary (I've been able to confirm this using postgres), the issue with PickleType is that Sqlalchemy pickles the objects automatically, despite having a json (or any but pickle) serializer configured for our app.

Hence, it seems that results from old task runs (tasks ran before this change) will have a pickled results instead of a json serialized one.

Can a celery client access to tasks results that were already ran in the past? I'm asking because am a celery noob.

Reply:
> Can a celery client access to tasks results that were already ran in the past?

Yes, this is possible if someone has access to the result (task) ID.

This is indeed a breaking change, for configurations that used a different serializer than the pickle serializer. Perhaps this can be solved by allowing a transitional fallback to `pickle.loads` on a deserialization error, but only with a configuration option. So the new serializer support will be disabled by default, and we emit a warning for the deprecation in the next version.

How does that sound?

Reply:
> Yes, this is possible if someone has access to the result (task) ID.

Alright then, let's suppose that this issue never existed and the serialization was working fine for any serializer. Also suppose someone started using celery configured with the json serializer, and after some time he changes his mind and decides to use pickle serializer. In this case, if someone has access to a result (task) ID corresponding to a task ran before the serializer change, would in this case the same deserialization error be expected?

Reply:
sqla should have support jsonfield as well, need to check that